testcontainers-dapr/src/test/java/io/dapr/testcontainers/DaprComponentTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2024 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.testcontainers;
+
+import io.dapr.testcontainers.DaprContainer.Component;
+import io.dapr.testcontainers.DaprContainer.Subscription;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Set;
+
+public class DaprComponentTest {
+
+  @Test
+  public void componentStateStoreSerializationTest() {
+    DaprContainer dapr = new DaprContainer("daprio/daprd")
+        .withAppName("dapr-app")
+        .withAppPort(8081)
+        .withComponent(new Component(
+            "statestore",
+            "state.in-memory",
+            "v1",
+            Collections.singletonMap("actorStateStore", new QuotedBoolean("true"))))
+        .withAppChannelAddress("host.testcontainers.internal");
+
+    Set<Component> components = dapr.getComponents();
+    Assert.assertEquals(1, components.size());
+
+    Component kvstore = components.iterator().next();
+    Assert.assertEquals(false, kvstore.getMetadata().isEmpty());
+
+    String componentYaml = dapr.componentToYaml(kvstore);
+    String expectedComponentYaml = "metadata:\n" + "  name: statestore\n"
+        + "apiVersion: dapr.io/v1alpha1\n"
```

Comment:
I was wondering if it would be cleaner to move these YAML files to `resources`, so we don't have to use these "ugly" string concatenations.

Reply:
Let's leave this for a `good-first-issue`, so we can help someone new to contribute.